pytorch_lightning/utilities/cli.py

```diff
@@ -153,107 +156,154 @@ def __init__(
-        self.prepare_fit_kwargs()
-        self.before_fit()
-        self.fit()
-        self.after_fit()
 
-    def init_parser(self) -> None:
+        self.run_subcommand()
+
+    def init_parser(self, use_base: bool = False, **kwargs: Any) -> LightningArgumentParser:
         """Method that instantiates the argument parser"""
-        self.parser = LightningArgumentParser(**self.parser_kwargs)
+        cls = ArgumentParser if use_base else LightningArgumentParser
+        return cls(**kwargs)
 
-    def add_core_arguments_to_parser(self) -> None:
-        """Adds arguments from the core classes to the parser"""
-        self.parser.add_argument(
+    def add_arguments(self, parser: LightningArgumentParser) -> None:
+        """
+        Implement to add extra arguments to the base parser or link arguments
+
+        Args:
+            parser: The base parser object to which arguments can be added
+        """
+        parser.add_argument(
             '--seed_everything',
             type=Optional[int],
             default=self.seed_everything_default,
             help='Set to an int to run seed_everything with this value before classes instantiation',
         )
-        self.parser.add_lightning_class_args(self.trainer_class, 'trainer')
-        trainer_defaults = {'trainer.' + k: v for k, v in self.trainer_defaults.items() if k != 'callbacks'}
-        self.parser.set_defaults(trainer_defaults)
-        self.parser.add_lightning_class_args(self.model_class, 'model', subclass_mode=self.subclass_mode_model)
-        if self.datamodule_class is not None:
-            self.parser.add_lightning_class_args(self.datamodule_class, 'data', subclass_mode=self.subclass_mode_data)
 
-    def add_arguments_to_parser(self, parser: LightningArgumentParser) -> None:
-        """Implement to add extra arguments to parser or link arguments
+    @property
+    def subcommands(self) -> Dict[str, List[str]]:
+        """Defines the list of available subcommands and the arguments to skip"""
+        return {
+            'fit': ['model', 'train_dataloader', 'val_dataloaders', 'datamodule'],
+            'validate': ['model', 'val_dataloaders', 'datamodule'],
+            'test': ['model', 'test_dataloaders', 'datamodule'],
+            'predict': ['model', 'dataloaders', 'datamodule'],
+            'tune': ['model', 'train_dataloader', 'val_dataloaders', 'datamodule'],
+        }
+
+    def add_subcommands(self, parser: LightningArgumentParser) -> None:
+        # TODO: default fit
+        parser_subcommands = parser.add_subcommands()
+        for subcommand in self.subcommands:
+            subcommand_parser = self.prepare_subcommand_parser(subcommand)
+            # TODO: add help
+            parser_subcommands.add_subcommand(subcommand, subcommand_parser)
+
+    def prepare_subcommand_parser(self, subcommand: str) -> LightningArgumentParser:
+        # TODO: pass env_prefix and default_env?
+        parser = self.init_parser(use_base=True)
```

Comment:
This parser object becomes useless once it gets added as a subcommand. Basically all of its actions are removed and added to the main parser. Options like `env_prefix` are ignored. I don't see the need to call `init_parser`.

```suggestion
        parser = ArgumentParser()
```

Reply:
Used it for max flexibility, but that's okay, changed it.